### DISCUSSION TITLE
Flags image not found

### DISCUSSION THREAD

**Discussion by PH5HP** [rejetto/hfs#720] [General]
Hi,

I noticed today that the flag images in the LOG section are not found anymore. Maybe somewhere something is missing, but I do not know. Any thoughts?

![image](https://github.com/user-attachments/assets/d576306c-1c4c-43a3-a6e4-d35b99d0f0da)

Kind regards,
Henk - https://fileserver.ph5hp.nl - https://ph5hp.nl

**Comment by PH5HP** [comment]
Hi,

Never mind. It was the code from the Hot Link Protection topic what caused it. I replaced it with your code, rejetto. Now everything is OK again.

Kind regards,
Henk - https://fileserver.ph5hp.nl/ - https://ph5hp.nl/

**Comment by rejetto** [comment]
Thank goodness :D 
i had already studied your problem without success

**Reply by PH5HP** [reply]
Hi,

This code was the cause:
------------------------------------------------------------------------------------------------------------------------------------------
/**
 * Middleware function to implement hotlink protection for image and video files.
 * Checks if the requested file is an image or video and if the Referer header is present and allowed.
 * If not, returns a 403 Forbidden status.
 */
exports.middleware = ctx => {
  // Check if the requested URL is an image or video file
  if (/\.(png|gif|jpe?g|mp4)$/i.test(ctx.url)) {
    const allowedReferers = ['your-domain.com', 'another-allowed-domain.com']; // Whitelist of allowed domains

    // Check if the Referer header is present and if it is in the whitelist
    if (!ctx.get('referer') || !allowedReferers.some(referer => ctx.get('referer').includes(referer))) {
      // If no Referer is present or it is not in the whitelist, set the status code to 403 (Forbidden)
      ctx.status = 403;
    }
  }
};

------------------------------------------------------------------------------------------------------------------------------------------
It is the code from the 'Hot Link Protection'thread'. I replaced that code with your code in that thread and the image is back. Maybe make a plugin for that?

Kind regards,
Henk - https://fileserver.ph5hp.nl/ - https://ph5hp.nl/

**Comment by rejetto** [comment]
what are you trying to achieve exactly?

**Reply by PH5HP** [reply]
Hi,

I thought to use that code to prevent direct linking to the files. But I am not a great programmer in javascript (yet) and used the code provided in that thread.

Kind regards,
Henk - https://fileserver.ph5hp.nl/ - https://ph5hp.nl/

**Comment by rejetto** [comment]
what do you mean by "direct" linking?
make an example please

**Reply by PH5HP** [reply]
Hi,

The same as beeing discussed in topic:
 https://github.com/rejetto/hfs/discussions/711

Kind regards,
Henk - https://fileserver.ph5hp.nl/ - https://ph5hp.nl/

**Comment by rejetto** [comment]
> Maybe make a plugin for that?

maybe in the future, but anyone feel free to do it.
What I want to do is to let people add multiple server_code-s,
as if you (anyone) need more than one, you probably don't know how to "merge" them into one.
